core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -282,6 +316,114 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
     assert(!failedTaskSet)
   }
 
+  test("scheduled tasks obey task and stage blacklists") {
```

Comment:
sure.  to be honest, I'm not sure how much it helps -- pulling out the common setup makes it a bit harder to trace what is going on, IMO.  but I also don't feel strongly.  Its the last commit so you can see it both ways:

https://github.com/apache/spark/pull/15644/commits/6a026341de96c67218d18421a69a780a04994112